pyramid_swagger/api.py

```diff
@@ -266,23 +266,49 @@ def _get_target_url(spec, target, current_path=''):
         )
 
 
+def _ensure_sane_key(fragment, mar_target):
+    '''
+    This function patches a nasty bug with the
+    'allOff' keyword in a spec, our resolved
+    ref keys become corrupted.
+
+    explanation:
+        sometimes the fragment becomes:
+        '/definitions/:........swagger|..definitions..A'
+        and which makes the marshaled_target:
+        ':........swagger|..definitions..:........swagger|..definitions..A'
```

Comment:
I don't want to mess with the actual recursion too much because I don't know who else uses this code and I wouldn't want to change it in any way :P